crypto/provider_core.c

```diff
@@ -683,6 +686,8 @@ void ossl_provider_free(OSSL_PROVIDER *prov)
         if (ref == 0) {
             if (prov->flag_initialized) {
                 ossl_provider_teardown(prov);
+                if (prov->cleanup_function != NULL)
+                    prov->cleanup_function((OSSL_CORE_HANDLE *)prov);
```

Comment:
I'm struggling to see the difference between the provider's teardown function, and the provider's cleanup_function?